x-pack/plugins/upgrade_assistant/server/saved_object_types/reindex_operation_saved_object_type.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsType } from 'src/core/server';
+
+import { REINDEX_OP_TYPE } from '../../common/types';
+
+export const reindexOperationSavedObjectType: SavedObjectsType = {
+  name: REINDEX_OP_TYPE,
+  hidden: false,
+  namespaceType: 'agnostic',
+  mappings: {
```

Comment:
The new Saved Object types API disallows dynamic mapping, which the original type definitions depended upon, so I had to add a bunch of new mappings to get the tests passing.

Reply:
If dynamic mappings in SO type are really a need, we can discuss about enabling it (actually it's only 'disabled' in the TS type, forcing the `dynamic: true` value does work, see (but please don't do):

https://github.com/elastic/kibana/blob/d72de0ea163ae4c5b9d454d5c7f91a7e2e835ecb/src/core/server/ui_settings/saved_objects/ui_settings.ts#L27-L30

It's just that when we discussed about it, we did not find any valid argument to allow this value on the public API as the SO type mapping is meant to be exhaustive.

Reply:
Thanks Pierre! I don't think we need it in this case. I also created https://github.com/elastic/kibana/issues/64547 so that we eventually remove these added mappings (if possible).